Cilsil/Cil/Parsers/ExceptionParser.cs

```diff
@@ -0,0 +1,293 @@
+                                                   fieldType,
+                                                   state.CurrentLocation));
+                state.PushExpr(exceptionVariable, fieldType);
+            }
+            else{
+                state.AppendToPreviousNode = true;
+            }
+
+            RegisterNode(state, newNode);
+             
+            // The first copy of this stack item was popped in the invocation of the
+            // constructor, so we push another on.
+            state.PushExpr(new VarExpression(fieldIdentifier), fieldType);
+            state.PushInstruction(instruction, newNode);
+
+            // Append the next instruction (which should be stloc, for representing
+            // storage of the constructed object into a local variable) to this new node.
+            instruction = state.PopInstruction();
+            ParseCilInstruction(instruction, state);
+
+            return true; 
+        }        
+
+        /// <summary>
+        /// Creates a exception method call returned via out parameter.
+        /// </summary>
+        /// <param name="state">Current program state.</param>
+        /// <param name="exceptedExpression">The excepted expression.</param>
+        /// <param name="exceptedExpressionType">The type of excepted expression.</param>
+        /// <param name="returnType">The return type of the exception method being called.</param>
+        /// <param name="returnVariable">Identifies the variable returned by the method.</param>
+        /// <param name="callArgs">The method arguments.</param>
+        /// <param name="exceptionCall">The exception Call SIL instruction.</param>
+        private void CreateExceptionCall(ProgramState state,
+                                         Expression exceptedExpression,
+                                         Typ exceptedExpressionType,
+                                         out TypeReference returnType,
+                                         out Identifier returnVariable,
+                                         out List<Call.CallArg> callArgs,
+                                         out Call exceptionCall)
+        {
+            callArgs = new List<Call.CallArg>();
+            try
+            {
+                returnType = state.OffsetToExceptionType[state.CurrentInstruction.Offset];
+            }   
+            catch(KeyNotFoundException)
+            {
+                returnType = state.Method.Module.TypeSystem.Object;
+            }
+            
+            var funcExp = new ConstExpression(ProcedureName.BuiltIn__unwrap_exception);
+            callArgs.Add(new Call.CallArg(exceptedExpression, exceptedExpressionType));
+            var callFlags = new Call.CallFlags(false, false, false);
+            returnVariable = state.GetIdentifier(Identifier.IdentKind.Normal);
+
+            if (returnType != state.Method.Module.TypeSystem.Object)
```

Comment:
style nit -- please use brackets around if blocks, even if it's 1 line.

Reply:
Done